qiskit_nature/transformers/second_quantization/electronic/seniority_zero_transformer.py

```diff
@@ -0,0 +1,246 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Seniority-Zero Restriction interface."""
+
+from copy import deepcopy
+import logging
+
+from functools import partial
+from typing import cast, Tuple
+import numpy as np
+
+from qiskit_nature import QiskitNatureError, ListOrDictType, settings
+from qiskit_nature.operators.second_quantization import VibrationalOp
+from qiskit_nature.properties import GroupedProperty, Property
+from qiskit_nature.properties.second_quantization import (
+    GroupedSecondQuantizedProperty,
+    SecondQuantizedProperty,
+)
+from qiskit_nature.properties.second_quantization.driver_metadata import DriverMetadata
+from qiskit_nature.properties.second_quantization.electronic import (
+    ParticleNumber,
+    ElectronicEnergy,
+)
+from qiskit_nature.properties.second_quantization.electronic.bases import (
+    ElectronicBasisTransform,
+    ElectronicBasis,
+)
+from qiskit_nature.properties.second_quantization.electronic.types import GroupedElectronicProperty
+from qiskit_nature.transformers.second_quantization import BaseTransformer
+
+logger = logging.getLogger(__name__)
+
+
+class SeniorityZeroTransformer(BaseTransformer):
+    """The Seniority-Zero restriction."""
```

Comment:
This class docstring/description of the transformer, will end up in the API reference as documentation of this. I think for many users this may not be well known and could do with a description, link(s) to paper(s) or other useful material. I see some method has a link but having things in the main docstring here will be helpful in an overall description. And perhaps examples/use cases of when this would be useful to consider using.

The second q ops creation, this now requires a use different qubit mappers right, and the fermionic ones, such as JW, should not be used. If so I think this aspect should be noted here about the change to the operators in that respect.